File: p006/p006.go
package main

import "fmt"
import "math"

func sumA(xs []int) int {
	s := 0
	for _, x := range xs {
		s += x
	}
	return int(math.Pow(float64(s), 2))
}

func sumB(xs []int) int {
	s := 0
	for _, x := range xs {
		s += int(math.Pow(float64(x), 2))
	}
	return s
}

func diff(n int) int {
	xs := make([]int, n)
	for i := 0; i < n; i++ {
		xs[i] = 1 + i
	}
	return sumA(xs) - sumB(xs)
}

func main() {
	fmt.Printf("%d\n", diff(100))
}

File: p006/p006.cpp
#include <iostream>
#include <math.h>
using namespace std;


int sumA(int xs[], int len) {
    int s = 0;
    for(int i=0; i<len; i++) {
        s += xs[i];
    }
    return (int)pow(s, 2);
}

int sumB(int xs[], int len) {
    int s = 0;
    for(int i=0; i<len; i++) {
        s += (int)pow(xs[i], 2);
    }
    return s;
}

int diff(int n) {
    int xs[n];
    for(int i=0; i<n; ++i) {
        xs[i] = 1 + i;
    }
    return sumA(xs, n) - sumB(xs, n);
}

int main() {
    cout << diff(100) << endl;
    return 0;
}
